text/0085-api-keys.md

```diff
@@ -0,0 +1,450 @@
+| API Key            | API keys are stored and managed from the endpoint `/keys` by the master key holder. These are the keys used by the technical teams to interact with MeiliSearch at the level of the client code. |
+
+### IV. Personas
+
+| Persona | Role |
+|---------|------|
+| Anna    | Anna has an `ssh` access to the MeiliSearch instance and manages it on a daily basis. |
+| Mark    | Mark is a developer from the same company as Anna. He will implement the code to communicate with MeiliSearch to solve technical/product needs. |
+
+### V. `API Key` Explanations
+
+#### Summary Key Points
+
+- `/keys` management is restricted to the master key.
+- When a master key is set at MeiliSearch first-launch, we generate two pre-configured `API Key` resources. A `Default Search API Key` restricted to the search action and a `Default Admin API Key` to handle all operations (except managing API Keys) on MeiliSearch.
+- New endpoints are added to manage the `API Key` resource.
+- `API keys` can have restrictions on which methods can be accessed via an `actions` list; they can also `expiresAt` a specific date time and be restricted to a specific set of `indexes`.
+
+#### Managing `API Key`
+
+![](https://i.imgur.com/mAUFnNb.png)
+
+`Anna` manages the MeiliSearch instance; she uses the master key she defined at startup to generate an `API Key` resource for `Mark` to use in his client code to communicate with the MeiliSearch instance.
+
+`Anna` can define access rights to certain indexes to define an expiration date and also authorized `actions` for an `API Key` (See API Key Actions List Definition Part).
+
+Only the master key allows managing the API keys.
+
+#### Master Key
+
+The master key exists to secure a MeiliSearch instance. As soon as a master key is set via the  `MEILI_MASTER_KEY` environment variable or the `--master-key` CLI option , the endpoint `/keys` is accessible only for the master key holder. It can be seen as a super admin key; It must be shared only with people who have to manage the security of a MeiliSearch instance.
+
+This master key is not an API Key, thus is not stored and fetchable from an API endpoint. It must be seen as a runtime lock that activates the security of MeiliSearch as soon as an instance is launched with it.
+
+At the first launch of MeiliSearch with a master key, MeiliSearch automatically generate two default API keys (See the `GET - /keys` example) to cover many of the most basic needs. It generates a `Default Search API Key` dedicated to the search that can be used on the client-side and a `Default Admin API Key` to manipulate a MeiliSearch instance from a backend side.
+
+If the value of the master key changes, the previously generated `API Keys` do not change or expire. Thus, MeiliSearch does not re-generate the previous generated default API keys.
```

Comment:
If we relaunch meilisearch with a new master key the private and public key remain the same?

Reply:
Yes, the `Default Search API Key` and the `Default Admin API Key` remain the same to add extra work by updating all clients using these keys.

Reply:
Can you change them in any way?

Reply:
Yes, they are editable!

Reply:
How can you edit them?